components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ThrottlingApiServiceImpl.java

```diff
@@ -907,6 +997,323 @@ public Response throttlingPoliciesCustomRuleIdDelete(String ruleId,  MessageCont
+            RestApiUtil.handleInternalServerError(errorMessage, log);
+        }
+        return resolveUpdateThrottlingPolicy(policyType, overwrite, exportThrottlePolicyDTO, messageContext);
+    }
+
+    /**
+     * @param policyType              Throttling policy type
+     * @param overwrite               User can either update an existing throttling policy with the same name or let the conflict happen
+     * @param exportThrottlePolicyDTO the policy to be imported
+     * @return Response with  message indicating the status of the importation and the imported/updated policy name
+     */
+    private Response resolveUpdateThrottlingPolicy(String policyType, boolean overwrite,
+            ExportThrottlePolicyDTO exportThrottlePolicyDTO, MessageContext messageContext)
+            throws APIManagementException {
+        ObjectMapper mapper = new ObjectMapper();
+        APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
+        String username = RestApiCommonUtil.getLoggedInUsername();
+        if (policyType.equals(RestApiConstants.RESOURCE_SUBSCRIPTION_POLICY)) {
+            SubscriptionThrottlePolicyDTO subscriptionPolicy = mapper.convertValue(exportThrottlePolicyDTO.getData(),
+                    SubscriptionThrottlePolicyDTO.class);
+            try {
+                Policy policyIfExists = apiProvider.getSubscriptionPolicy(username, subscriptionPolicy.getPolicyName());
+                if (policyIfExists != null) {
+                    if (overwrite) {
+                        String uuid = policyIfExists.getUUID();
+                        Response resp = throttlingPoliciesSubscriptionPolicyIdPut(uuid,
+                                RestApiConstants.APPLICATION_JSON, subscriptionPolicy, messageContext);
+                        String message = "Successfully updated Subscription Throttling Policy : "
+                                + subscriptionPolicy.getPolicyName();
+                        return Response.fromResponse(resp).entity(message).build();
+                    } else {
+                        RestApiUtil.handleResourceAlreadyExistsError(
+                                "Subscription Policy with name " + subscriptionPolicy.getPolicyName()
+                                        + " already exists", log);
+                    }
+                } else {
+                    Response resp = throttlingPoliciesSubscriptionPost(RestApiConstants.APPLICATION_JSON,
+                            subscriptionPolicy, messageContext);
+                    String message = "Successfully imported Subscription Throttling Policy : "
+                            + subscriptionPolicy.getPolicyName();
+                    return Response.fromResponse(resp).entity(message).build();
+                }
+            } catch (PolicyNotFoundException ignore) {
+            }
+        } else if (policyType.equals(RestApiConstants.RESOURCE_APP_POLICY)) {
```

Comment:
swap the equalling

Reply:
Fixed with 30d0ecb9c29c0024fa182ba597c8eda4c128c451